File: backend/app/schemas/wfh.py
"""Pydantic v2 DTOs — Work From Home requests."""
from __future__ import annotations
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel, Field


class ApplyWfhRequest(BaseModel):
    from_date: date
    to_date: date
    reason: str = Field(..., min_length=1, max_length=2000)


class WfhApprovalActionRequest(BaseModel):
    remarks: Optional[str] = None


class WfhRejectRequest(BaseModel):
    reason: str = Field(..., min_length=1, max_length=1000)


class WfhEmployeeSummary(BaseModel):
    id: int
    employee_code: str
    first_name: str
    last_name: str
    full_name: str
    email: str
    department_name: Optional[str] = None

    model_config = {"from_attributes": True}


class WfhResponse(BaseModel):
    id: int
    employee_id: int
    employee: Optional[WfhEmployeeSummary] = None
    from_date: date
    to_date: date
    reason: str
    status: str
    approver_id: Optional[int]
    approved_at: Optional[datetime]
    rejection_reason: Optional[str]
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: admin-panel/src/api/wfh.ts
import apiClient from "./client";

export interface WfhItem {
  id: number;
  employee_id: number;
  employee?: {
    id: number;
    employee_code: string;
    first_name: string;
    last_name: string;
    full_name: string;
    email: string;
    department_name?: string;
  };
  from_date: string;
  to_date: string;
  reason: string;
  status: string;
  approver_id?: number;
  approved_at?: string;
  rejection_reason?: string;
  created_at: string;
  updated_at: string;
}

export const fetchPendingWfh = async (): Promise<WfhItem[]> => {
  const response = await apiClient.get("/wfh/pending");
  return response.data;
};

export const fetchAllWfh = async (params: {
  company_id: number;
  page?: number;
  page_size?: number;
  department_id?: number;
  status?: string;
  from_date?: string;
  to_date?: string;
  search?: string;
}) => {
  const response = await apiClient.get("/wfh", { params });
  return response.data;
};

export const approveWfh = async (id: number) => {
  const response = await apiClient.post(`/wfh/${id}/approve`);
  return response.data;
};

export const rejectWfh = async (id: number, reason: string) => {
  const response = await apiClient.post(`/wfh/${id}/reject`, { reason });
  return response.data;
};
